idaes/generic_models/properties/core/eos/eos_base.py

```diff
@@ -51,6 +51,27 @@ def calculate_scaling_factors(b, pobj):
     def build_parameters(b):
         raise NotImplementedError(_msg(b, "build_parameters"))
 
+    @staticmethod
+    def get_vol_mol_pure(b, phase, comp, temperature):
+        try:
+            vol_mol = get_method(b, "vol_mol_"+phase+"_comp", comp)(
+                                 b, cobj(b, comp), temperature)
+        except (AttributeError, ConfigurationError):
+            # vol_mol not defined, try for dens_mol instead
+            try:
+                vol_mol = 1/get_method(b, "dens_mol_"+phase+"_comp", comp)(
```

Comment:
does dens_mol have the phase in the name? i.e. dens_mol_liq_comp? Why doesn't it just follow the other naming convention like dens_mol_phase_comp['liq', j]?

Reply:
This is a pure component property, not a mixture property. I.e. we are looking up the config dict for how to calculate dens_mol for a given component and phase type. This is then used to calculate `dens_mol_phase_comp` for the mixture.

Reply:
After our discussion, I learned that these are pure component parameters. 👍